modules/library/metadata/src/main/java/org/geotools/util/DateTimeParser.java

```diff
@@ -0,0 +1,634 @@
+package org.geotools.util;
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2019, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+import java.text.ParseException;
+import java.text.ParsePosition;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.TreeSet;
+import java.util.logging.Logger;
+import org.geotools.util.logging.Logging;
+
+/**
+ * Parses the {@code time} parameter of the request. The date, time and period are expected to be
+ * formatted according ISO-8601 standard.
+ *
+ * @author Cedric Briancon
+ * @author Martin Desruisseaux
+ * @author Simone Giannecchini, GeoSolutions SAS
+ * @author Jonathan Meyer, Applied Information Sciences, jon@gisjedi.com
+ * @author Daniele Romagnoli, GeoSolutions SAS
+ * @version $Id$
+ */
+public class DateTimeParser {
```

Comment:
This class is the merge of code and functionalities from GeoServer gs-ows TimeParser (being now reduced to a simple wrapper as part of https://github.com/geoserver/geoserver/pull/3884/files) and ImageMosaic's TimeParser utility class